src/cryptsetup/cryptsetup-tokens/cryptsetup-token-util.c

```diff
@@ -0,0 +1,47 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "cryptsetup-token-util.h"
+#include "string-util.h"
+
+int crypt_dump_buffer_to_hex_string(
+                const char *buf,
+                size_t buf_size,
+                char **ret_dump_str) {
+
+        int r;
+
+        assert(ret_dump_str);
+
+        for (unsigned int i = 0; i < buf_size; i++) {
+                /* crypt_dump() breaks line after every
+                 * 16th couple of chars in dumped hexstring */
+                r = strextendf_with_separator(
+                        ret_dump_str,
```

Comment:
so ret_dump_str is both input and output parameter. Which is OK, but then it shouldn't be prefixed with "ret_", because that's how we name parameters that are strictly for returning stuff and that we only write to but don't read from.